### PR TITLE
feat: #11357 payments fields updates

### PR DESCRIPTION
**Pull request by willmcvay**
# Pull request checklist

**Detail as per issue below (required):**

fixes: #11357

### CODE REVIEW

File: packages/payments-client/src/components/payment/payment.tsx [hunk 1]
```diff
@@ -174,13 +173,13 @@ export const Payment: FC<PaymentProps> = ({ paymentRequest }) => {
...
-      Authorization: idToken,
-      'reapit-customer': clientCode as string,
+      'reapit-customer': clientCode,
+      'reapit-id-token': idToken,
       'reapit-app-id': process.env.appId,
     },
   })
 
-  useEffect(handleGetMerchantKey(getMerchantKey, idToken, config), [config, idToken])
+  useEffect(handleGetMerchantKey(getMerchantKey, idToken, config), [idToken, config])
```
Comment (bashleigh) [inline packages/payments-client/src/components/payment/payment.tsx:182]: Just checking, did you mean to swap these values here?

File: packages/payments-service/src/core/authorizer.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  if (!idToken) {
+    throw new Error('No idToken provided')
+  }
+
+  const decodedIdToken = jwt.decode(idToken, { complete: true })
+
+  if (!decodedIdToken) throw new Error('idToken failed to decode')
+  if (typeof decodedIdToken.payload === 'string') throw new Error('Decoded idToken payload is a string')
+  if (!decodedIdToken.payload.sub) throw new Error('idToken does not contain a sub')
+  if (!decodedIdToken.payload.aud) throw new Error('idToken does not contain an aud')
```
Comment (bashleigh) [inline packages/payments-service/src/core/authorizer.ts:33]: Do these errors throw a 500 or 401 etc? Guessing I can check in `./../../../ts-scripts/src/authorizer/index`?

File: packages/payments-client/src/components/payment/payment.tsx (hunk 1 above)
Comment (willmcvay) [reply packages/payments-client/src/components/payment/payment.tsx:182]: I removed the idToken then had to add it back, long day!

File: packages/ts-scripts/src/authorizer/index.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import * as jwt from 'jsonwebtoken'
+import { JwtRsaVerifier, CognitoJwtVerifier } from 'aws-jwt-verify'
+
+export const authorizerHandler =
+  (customChallenge?: (event: APIGatewayTokenAuthorizerEvent, decodedToken: jwt.Jwt) => Promise<void | Error>) =>
+  async (event: APIGatewayTokenAuthorizerEvent): Promise<APIGatewayAuthorizerResult> => {
+    try {
+      if (!event['headers'].authorization) throw new Error('No authorization header provided')
+
+      const token = event['headers'].authorization.split('Bearer ')[1]
```
Comment (bashleigh) [inline packages/ts-scripts/src/authorizer/index.ts:11]:
```suggestion
      const token = event['headers'].authorization.split('Bearer ').pop()
```

File: packages/payments-service/src/core/authorizer.ts (hunk 2 above)
Comment (willmcvay) [reply packages/payments-service/src/core/authorizer.ts:33]: It gets caught then the authorizer returns a deny policy, yep. I kinda want that error to bubble

**Review by bashleigh — APPROVED** [review]
Approved

File: packages/payments-client/src/components/payment/payment.tsx (hunk 1 above)
Comment (willmcvay) [reply packages/payments-client/src/components/payment/payment.tsx:182]: It's just a useEffect dependency array - the order doesn't matter

**Pull request merged**